### PR TITLE
[PILOT-1560]: Improve GET datasets endpoint to return list of datasets filtered by list of IDs

### PR DESCRIPTION
**Pull request by hugoantunes**
## Summary

add filter by ID to get datasets endpoint

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1560

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes
- [ ] No

## Test Directions

(Additional instructions for how to run tests or validate functionality if not covered by unit tests)

### CODE REVIEW

File: tests/components/dataset/test_views.py [hunk 1]
```diff
@@ -124,3 +124,14 @@ async def test_list_datasets_returns_datasets_filtered_by_created_at_start_and_c
 
         assert received_ids == [str(dataset.id)]
         assert received_total == 1
+
+    async def test_list_datasets_returns_dataset_filtered_by_ids(self, client, jq, dataset_factory):
```
Comment (mloshakov) [inline tests/components/dataset/test_views.py:128]: Please add tests with lists of incorrect values, either not a valid UUIDs or non-existent ones.

**Review by mloshakov — CHANGES_REQUESTED** [review]
I have one small improvement suggestion

File: tests/components/dataset/test_views.py (hunk 1 above)
Comment (hugoantunes) [reply tests/components/dataset/test_views.py:128]: added

**Review by mloshakov — APPROVED** [review]
👍

**Pull request merged**